subsys/suit/ipuc/src/suit_ipuc_sdfw.c

```diff
@@ -0,0 +1,695 @@
+		suit_ipuc_entry_t *p_entry = &components[i];
+
+		if (suit_compare_zcbor_strings(component_id, &p_entry->component_id)) {
+			return p_entry;
+		}
+	}
+
+	return NULL;
+}
+
+static suit_ipuc_entry_t *ipuc_entry_from_handle(suit_component_t handle)
+{
+	struct zcbor_string *component_id = ipuc_component_id_from_handle(handle);
+
+	return ipuc_entry_from_component_id(component_id);
+}
+
+static suit_plat_err_t ipuc_decode_mem_component_id(struct zcbor_string *component_id,
+						    intptr_t *slot_address, size_t *slot_size)
+{
+	suit_component_type_t component_type = SUIT_COMPONENT_TYPE_UNSUPPORTED;
+	intptr_t decoded_slot_address = 0;
+	size_t decoded_slot_size = 0;
+
+	if (suit_plat_decode_component_type(component_id, &component_type) != SUIT_PLAT_SUCCESS ||
+	    component_type != SUIT_COMPONENT_TYPE_MEM) {
+		return SUIT_PLAT_ERR_UNSUPPORTED;
+	}
+
+	if (suit_plat_decode_address_size(component_id, &decoded_slot_address,
+					  &decoded_slot_size) != SUIT_PLAT_SUCCESS ||
+	    decoded_slot_address == 0 || decoded_slot_size == 0) {
+		return SUIT_PLAT_ERR_UNSUPPORTED;
+	}
+
+	*slot_address = decoded_slot_address;
+	*slot_size = decoded_slot_size;
+
+	return SUIT_PLAT_SUCCESS;
+}
+
+suit_plat_err_t suit_ipuc_sdfw_get_count(size_t *count)
+{
+
+	if (count == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	size_t declared_count = 0;
+
+	k_mutex_lock(&ipuc_mutex, K_FOREVER);
+	for (size_t i = 0; i < CONFIG_SUIT_IPUC_SIZE; i++) {
+		suit_ipuc_entry_t *ipuc_entry = &components[i];
+
+		if (ipuc_entry->component_id.value != NULL && ipuc_entry->component_id.len != 0) {
+			declared_count++;
+		}
+	}
+	*count = declared_count;
+	k_mutex_unlock(&ipuc_mutex);
+	return SUIT_PLAT_SUCCESS;
+}
+
+suit_plat_err_t suit_ipuc_sdfw_get_info(size_t idx, struct zcbor_string *component_id,
+					suit_manifest_role_t *role)
+{
+	if (component_id == NULL || role == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	size_t cur_idx = 0;
+
+	k_mutex_lock(&ipuc_mutex, K_FOREVER);
+	for (size_t i = 0; i < CONFIG_SUIT_IPUC_SIZE; i++) {
+		suit_ipuc_entry_t *ipuc_entry = &components[i];
+
+		if (ipuc_entry->component_id.value != NULL && ipuc_entry->component_id.len != 0) {
+			if (cur_idx == idx) {
+				component_id->value = ipuc_entry->component_id.value;
+				component_id->len = ipuc_entry->component_id.len;
+				*role = ipuc_entry->role;
+				k_mutex_unlock(&ipuc_mutex);
+				return SUIT_PLAT_SUCCESS;
+			}
+			cur_idx++;
+		}
+	}
+
+	k_mutex_unlock(&ipuc_mutex);
+	return SUIT_PLAT_ERR_NOT_FOUND;
+}
+
+suit_plat_err_t suit_ipuc_sdfw_write_setup(int ipc_client_id, struct zcbor_string *component_id,
+					   struct zcbor_string *encryption_info,
+					   struct zcbor_string *compression_info)
+{
+	intptr_t slot_address = 0;
+	size_t slot_size = 0;
+
+	if (component_id == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	if (ipuc_decode_mem_component_id(component_id, &slot_address, &slot_size) !=
+	    SUIT_PLAT_SUCCESS) {
+		/* Only MRAM-based MEM components are supported
```

Comment:
Looking at the code of the `ipuc_decode_mem_component_id` there is no place where we verify the the address range is in MRAM. A MEM component can point to an address in MRAM, RAM or extmem.

You should probably add a call to `suit_memory_global_address_is_in_nvm`